### PR TITLE
Added Advanced Search using Groups

### PR DESCRIPTION
**Pull request by weissm**
The  [Group API](https://docs.gitlab.com/ee/api/search.html#scope-projects-1) specifies the following format:

> `curl --header "PRIVATE-TOKEN: <your_access_token>" "https://gitlab.example.com/api/v4/groups/3/search?scope=projects&search=flight"`.

This group selection is supported by introducing the `groupID `parameter to the `ProjectQuery` scope.

It addresses the requests in the former pull request: `https://github.com/ubisoft/NGitLab/pull/129`. All tests are passing except for unrelated assert in `Test_merge_request_approvers`.

### CODE REVIEW

File: NGitLab/IGroupsClient.cs [hunk 1]
```diff
@@ -29,6 +29,8 @@ public interface IGroupsClient
 
         IEnumerable<Project> SearchProjects(int groupId, string search);
 
+        IEnumerable<Project> SearchProjects(string groupId, string search, GroupQueryScope scope);
```
Comment (meziantou) [inline NGitLab/IGroupsClient.cs:32]: When we add new overloads in the interface, we try to use a class. This way, we don't need to re-update it if we want to add a new parameter.
Also, as this is a new class, you should use properties (`{get;set}`) instead of fields.

```suggestion
        IEnumerable<Project> SearchProjects(SearchProjectQuery query);
```

File: NGitLab/Impl/GroupsClient.cs [hunk 2]
```diff
@@ -91,6 +91,16 @@ public IEnumerable<Project> SearchProjects(int groupId, string search)
             return _api.Get().GetAll<Project>(url);
         }
 
+        // implementation according to https://docs.gitlab.com/ee/api/search.html#scope-projects-1
+        public IEnumerable<Project> SearchProjects(string groupId, string search, GroupQueryScope scope)
+        {
+            var url = Url + "/" + Uri.EscapeDataString(groupId.ToString(CultureInfo.InvariantCulture)) + "/search";
+            url = Utils.AddParameter(url, "scope", Enum.GetName(typeof(GroupQueryScope), scope).ToLower());
```
Comment (meziantou) [inline NGitLab/Impl/GroupsClient.cs:98]:
```suggestion
            url = Utils.AddParameter(url, "scope", scope.ToString().ToLowerInvariant());
```

File: NGitLab/IGroupsClient.cs (hunk 1 above)
Comment (meziantou) [reply NGitLab/IGroupsClient.cs:32]: According to the documentation examples, changing the `scope` also change the result type. For instance, `scope=merge_requests` returns a list of merge request. If so, we cannot allow any scope and return a list of projects.

File: NGitLab/Models/GroupQuery.cs [hunk 3]
```diff
@@ -62,5 +63,56 @@ public class GroupQuery
         /// (optional)
         /// </summary>
         public AccessLevel? MinAccessLevel;
+
+        public GroupQueryScope Scope = GroupQueryScope.Projects;
```
Comment (meziantou) [inline NGitLab/Models/GroupQuery.cs:67]: As it is required, let the user set it explicitly

```suggestion
        public GroupQueryScope? Scope;
```

File: NGitLab/NGitLab.csproj [hunk 4]
```diff
@@ -2,6 +2,9 @@
   <PropertyGroup>
     <TargetFrameworks>net45;netstandard2.0</TargetFrameworks>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net45|AnyCPU'">
```
Comment (meziantou) [inline NGitLab/NGitLab.csproj:5]: I think you should revert this change

**Review by meziantou — CHANGES_REQUESTED** [review]

**Comment by weissm** [comment]
Incorporated comments in https://github.com/ubisoft/NGitLab/pull/131

File: NGitLab/IGroupsClient.cs (hunk 1 above)
Comment (weissm) [reply NGitLab/IGroupsClient.cs:32]: Very valid comment. Shall we reduce now the GroupQueryScope options again to allow only the proper return types?

Comment (meziantou) [reply NGitLab/IGroupsClient.cs:32]: If I'm not mistaken, the only valid option (to return a collection of projects) is `project`, but there is already this overload that does exactly that `IEnumerable<Project> SearchProjects(int groupId, string search);`
